### ISSUE TITLE
Call is ended during Dialing/Ringing if mute microphone is pressed

### ISSUE THREAD

**Issue by guimspace** [signalapp/Signal-Android#9421]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
The call is ended if the button "mute micrphone" is pressed during "dialing" or "ringing".

This not happen in Signal for iOS.

### Steps to reproduce
1. Start a call or video call
2. On the Signal Call scree, press "mute microphone"

**Actual result:** The call is ended with message "Ending call".
**Expected result:** The call is not ended, the microphone stays muted.

### Device info
**Device:** ASUS ZenFone 3 ZE520KL
**Android version:** 8.0.0
**Signal version:** 4.56.1

### Link to debug log

https://debuglogs.org/1294ded07943048d692c39920cce6b3ca214a42beadf177c6635270f5504b25f

**Comment by frankvollebregt** [comment]
I have created a PR that resolves this issue. A check is added to ensure the call is not currently in the `DIALING` CallState.

**Comment by frankvollebregt** [comment]
This new PR additionally checks for the `REMOTE_RINGING` CallState, as this would still close the call in the initial PR.

**Comment by guimspace** [comment]
Hi @frankvollebregt. Thanks for your PR!

By the title, are you proposing to disable the mute button during the ring? I think it is best to keep it enabled while working as expected: mute the microphone while keeping the process of ring, and the microphone is still muted when the person answer the call.

**Comment by frankvollebregt** [comment]
Hey @guimspace. That may have been an unclear title then ;)

When the mute button is pressed (both before and after this PR) the microphone is always muted/unmuted, no matter the state of the call. After that, a check was in place that was supposedly meant to say
```if the call is ongoing then do something with it```
but this check did not work properly (either because it has always been like that, or caused by a change in Signal's Ringrtc repository that is used for the calls). As a result, the if statement would succeed even when the call was not active yet, and the call activity would crash, causing the call to terminate (hence the issue). I have updated this if statement to ensure it does not pass while the call is `DIALING` or `REMOTE_RINGING`.

**Comment by frankvollebregt** [comment]
tldr: The mute button still works, also while ringing

**Comment by guimspace** [comment]
The button doesn't work for me. When I press "mute" during ringing/calling, the call is ended.

**Comment by guimspace** [comment]
Oh, nerve mind.

> That may have been an unclear title then ;)

I thought you were referring to my title. So your PR is fixing the issue. Thanks!

**Comment by alex-signal** [comment]
Hello, this should be fixed in `4.59.1` beta.
